2-collectors/scala-stream-collector/core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorRoute.scala

```diff
@@ -33,7 +33,15 @@ trait CollectorRoute {
   def extractContentType: Directive1[ContentType] =
     extractRequestContext.map(_.request.entity.contentType)
 
-  def collectorRoute: Route =
+  def collectorRoute =
+    if (collectorService.allowRedirects) routes else rejectRedirect ~ routes
+
+  def rejectRedirect: Route =
+    path("r" / Segment) { _ =>
+      complete(StatusCodes.Gone -> "redirects disabled")
```

Comment:
Why `410` actually? 
> The HTTP specification intends this status code to be used for "limited-time, promotional services". APIs should not feel compelled to indicate resources that have been deleted with this status code.

Is this true for this redirects?
This is from https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

Maybe simple `404`, wdyt?

Reply:
Interesting, haven't heard about "limited-time, promotional offers" part. `404` makes sense.